dev-guide/api-conventions.md

```diff
@@ -279,6 +279,37 @@ myPlatformConfig:
 
 This is invalid. Multiple structs have been configured and no discriminant is provided.
 
+### Prefer consistency with Kubernetes style APIs
+
+As both Kubernetes and OpenShift have a number of integrations with cloud and infrastructure platforms, there are a
+number of different abstractions built into OpenShift that abstract and extend infrastructure capabilities.
+For example, the Machine API or platform specific storage integrations within OpenShift.
+
+These integrations often result in adding new API fields to OpenShift to allow users to configure various features of
+the platform. Often, different platform's concepts are similar, but their naming of particular features can be specific
+to their platform.
+If we copy verbatim the API from a platform, this may or may not be compliant with OpenShift conventions and may or
+may not make sense in a wider scope.
+
+If we intend to support similar features across platforms, it is preferable to have similar APIs for these features
+within OpenShift rather than mimicking the upstream API. This has the benefit of providing consistency to users when
```

Comment:
Here "upstream API" means the platform API?

Reply:
Yes it does, I'll swap